### PR TITLE
Fixed inappropriate initializations of "startOfLine" in readtags.c.

### PR DESCRIPTION
**Pull request by siegel**
The previous change was incompatible with systems on which fpos_t is not a scalar.

### CODE REVIEW

File: readtags.c [hunk 1]
```diff
@@ -415,7 +415,7 @@ static char *duplicate (const char *str)
 
 static void readPseudoTags (tagFile *const file, tagFileInfo *const info)
 {
-	fpos_t startOfLine = 0;
+	fpos_t startOfLine;
```
Comment (masatake) [inline readtags.c]: I will work on the initialization issue of `startofLine`.
I am reading https://www.securecoding.cert.org/confluence/display/c/FIO19-C.+Do+not+use+fseek%28%29+and+ftell%28%29+to+compute+the+size+of+a+regular+file and thinking its impact on ctags.

**Comment by masatake** [comment]
I guess you are rather new to git. If this is not the last time that you work on ctags, I strongly recommend to make "topic branch" for each hacking. You are using "master" branch. I recommend you not to use "master" branch for hacking. It is just for synchronizing with  the upstream.

https://git-scm.com/book/en/v2/Git-Branching-Branching-Workflows
https://git-scm.com/book/ch3-5.html

In addition I would like you to know "push --force" and squashing with "git rebase -i".

This PR has 5 commits. 4 commits of the 5 are your "trial and error". If possible, I would like to avoid including such "trial and error" commits; the log becomes hard to be read.


9434eb3 and 
8849b9c have been merged already.

So I would like to remove them from this pull request. 54b1b7b can be removed, too.


2526de6 and 12deac9 can be unified. As the result your pull request becomes one commit. (top half)
After the unification I would like you to split the commit into multiple smaller commits. (bottom half)

The top half
---------------------------------
0. add following lines to your ctags/.git/config
```
[remote "upstream"]
	url = https://github.com/fishman/ctags.git
	fetch = +refs/heads/*:refs/remotes/upstream/*
```

1. rename your branch for this topic.

```
    git branch -m "improve-readtags"
```

2. make new master branch

```
    git checkout upstream/master
    git branch master
    git checkout master
    git pull upstream master:master
```

3. rebsae the topic branch -i master

```
    git checkout "improve-readtags"
    git rebase master
```

You will get reports about conflicts. You have to solve them.

4. rearrange the topic branch

```
    git rebase -i master
```

5. make new pull request

```
    git push origin `improve-readtags:improve-readtags`
```

6. (My side) I will review the new pull request.

7. (Your side) reflect the comment
```
    git commit
```

8. (Your side) rearrange the commit
```
    git rebase -i master
```

The bottom half
------------------------------------------------------
```
-	Fixed a case in which certain search pattern formations in the
	tags file would not be located, and/or cause `readtags` to crash.

-	Fixed bug in which `-p` would fail on sorted tag files.

-	report an error and exit if the tag file doesn't contain at least
	a `!_TAG_PROGRAM_NAME` key.

-	Corrected typo of "Fileter" in the output of `-h`

-	usage output now includes the program's name as given in `argv[0]`
```

All tihngs are fixed in a commit. It is difficult for me to verify the changes.
Split the commit for each fixing.

> -	Fixed a case in which certain search pattern formations in the
	tags file would not be located, and/or cause `readtags` to crash.
Need a test case. What is certain search pattern?  This is really critical.
see http://docs.ctags.io/en/latest/tmain.html. Tmain/readtags-qualifier.d/ is an example of testing readtags.

> -	Fixed bug in which `-p` would fail on sorted tag files.
I would like to get a test case for this bug, too. Making a new pull request only for this fix is also acceptable.

> -	Corrected typo of "Fileter" in the output of `-h`
Thank you. Making a new pull request only for this fix is also acceptable.

**Comment by siegel** [comment]
> I guess you are rather new to git. 

You are correct. :-) I greatly appreciate the advice and guidance, and will try to find time soon to implement these changes and submit a fresh set of pull requests. (Work's a little busy at the moment…)

**Comment by masatake** [comment]
When I started working on ctags, I didn't know git well. Co-developers taught me:-)

 Small well separated pull requests are always welcome.

**Pull request closed without merging**